synapse/storage/database.py

```diff
@@ -920,13 +920,23 @@ def simple_insert_many_txn(
             if k != keys[0]:
                 raise RuntimeError("All items must have the same keys")
 
-        sql = "INSERT INTO %s (%s) VALUES(%s)" % (
-            table,
-            ", ".join(k for k in keys[0]),
-            ", ".join("?" for _ in keys[0]),
-        )
+        if isinstance(txn.database_engine, PostgresEngine):
+            # We use `execute_values` for postgres as it can be a lot faster
+            # than `execute_batch`, but it's only available on postgres.
```

Comment:
Seems we got a bit aggressive with the "is postgres" :) (I think this was copied below?)

```suggestion
            # We use `execute_values` as it can be a lot faster than `execute_batch`,
            # but it's only available on postgres.
```